### PR TITLE
Feat/email type

### PR DESCRIPTION
**Pull request by anveshthakur**
hi @plutov, 

I have added the new question type email which is mentioned in the issue #20, I've tested the code from the UI It is working fine. Please let me know if there is any change required or you want me to add any other validations to it.

Thanks!

### CODE REVIEW

File: api/pkg/types/answers.go [hunk 1]
```diff
@@ -151,3 +152,22 @@ func (a BoolAnswer) Value() (driver.Value, error) {
...
+type EmailAnswer struct {
+	AnswerValue string `json:"value"`
+}
+
+func (a EmailAnswer) Value() (driver.Value, error) {
+	return json.Marshal(a)
+}
+
+func (a *EmailAnswer) Validate(q Question) error {
+	re := regexp.MustCompile(`^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`)
```
Comment (plutov) [inline api/pkg/types/answers.go:165]: This is not ideal email validation. Let's use some package like ozzo/validation

File: api/surveys/short/questions.yaml [hunk 2]
```diff
@@ -10,4 +10,7 @@ questions:
   - Hamburg
   - Cologne
   - Geneva
-  - Oslo
\ No newline at end of file
+  - Oslo
+- type: email-text
```
Comment (plutov) [inline api/surveys/short/questions.yaml:14]: I'd prefer to call it just "email".
Also, we need to add docs.
Can you add it to "simple" instead of "short"?

**Review by plutov — CHANGES_REQUESTED** [review]

File: api/pkg/types/answers.go (hunk 1 above)
Comment (anveshthakur) [reply api/pkg/types/answers.go]: have added ozzo/validations to email let me know if you want me to add this to other answer type validators as well.

File: api/surveys/short/questions.yaml (hunk 2 above)
Comment (anveshthakur) [reply api/surveys/short/questions.yaml]: have changed email-text to email and added docs for the same, have moved the question from short to simple survey

**Comment by anveshthakur** [comment]
Hi @plutov,
I’ve added ozzo/validation to include proper checks on the email as requested. I accidentally marked the conversation as resolved, but I’m happy to make further changes if needed.

**Comment by plutov** [comment]
Thanks @anveshthakur the PR looks good. I will test it on my machine once I have a bit more time.

**Comment by anveshthakur** [comment]
sure @plutov, thanks!

**Comment by plutov** [comment]
Thanks @anveshthakur your change is now in the latest release!

**Comment by plutov** [comment]
@anveshthakur I only noticed that the tests have failed but fixed those.

**Comment by plutov** [comment]
@anveshthakur I also noticed that email is not shown on the response page.
Screenshot 2024-09-27 at 23 23 54

**Pull request merged**